### PR TITLE
refactor(transactions) Switch SignerListSet to new transaction scheme

### LINKED ISSUE #277: A SignerListSet that clear a SignerList throws errors on any page is shown
A SignerList can be cleared by calling SignerListSet with a quorum of 0.  There is a bug in the parsing of SignerListSet where it always expects SignerEntries to be an array.

This breaks the following pages:

- [Ledger](https://testnet.xrpl.org/ledgers/30736727) (page won't render)
- [Transaction](https://testnet.xrpl.org/transactions/AB715572A238301FA8F29F06726290B6FE46C92C3E09083C0348D2393C908627) (page won't render)
- [Account](https://testnet.xrpl.org/accounts/rfqdGwfvJa4aigQuUtSQ8w41Hwp3cNeaB5) (transactions tab won't render)

Will be fixed along with #205

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change
This PR primarily moves SignerListSet to the new transaction schema, hooks, typescript, and more tests in addition to the following:

- Allow transaction components to be more correctly typed
- Rename `.unsetregularkey` to `.unset` to make it more generic
apply to SetRegularKey and SignerListSet.
- Fix bug related to transactions that remove all signers #277
- Add clsx a small (228B) library for dynamic classNames

Fixes #277 and completes #205

### Context of Change

This change also allows each transaction to define its own fields and instructions.  It will help clean up the massive Instructions type.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

### CODE REVIEW

File: public/locales/en-US/translations.json [hunk 1]
```diff
@@ -101,7 +102,7 @@
   "the_channel_balance_is": "The channel balance claimed is",
   "payment_channel_closed_description": "The payment channel will be closed, any remaining balance will be returned to the source account",
   "set_signer_list_description": "It sets the minimum signer quorum at {{quorum}} with the following list of signers",
-  "deletee_signer_list_description": "bbb",
+  "unset_signer_list_description": "It removed all signers from the account",
```
Comment (mvadari) [inline public/locales/en-US/translations.json:105]: lmao

**Review by pdp2121 — APPROVED** [review]

File: public/locales/en-US/translations.json [hunk 2]
```diff
@@ -102,7 +103,7 @@
...
```
Comment (mvadari) [inline public/locales/en-US/translations.json:106]: lmao

File: src/containers/Accounts/AccountTransactionsTable/styles.scss [hunk 3]
```diff
@@ -164,7 +164,7 @@
     .partial-payment,
     .closed,
     .flag,
-    .unsetregularkey,
+    .unset,
```
Comment (mvadari) [inline src/containers/Accounts/AccountTransactionsTable/styles.scss:167]: Why not just have two different `unset`, one for SignerListSet and one for SetRegularKey?

**Review by mvadari — APPROVED** [review]

File: src/containers/Accounts/AccountTransactionsTable/styles.scss (hunk 3 above)
Comment (ckniffen) [reply src/containers/Accounts/AccountTransactionsTable/styles.scss:167]: There will be more unsets in the future.  For example domains and email hashes on accountset can be cleared out.  We dont current do anything with those.

**Review by pdp2121 — APPROVED** [review]

**Pull request merged**